tests/e2e/utils/kubeflow_installation.py

```diff
@@ -137,15 +141,19 @@ def build_component(
             if component_name == "cert-manager":
                 print("wait for 30s for cert-manager-webhook resource to be ready...")
                 time.sleep(30)
-
-    if "validations" in path_dic[component_name]:
-        identifier = path_dic[component_name]["validations"]["identifier"]
-        common_label = path_dic[component_name]["validations"]["common_label"]
-        namespace = path_dic[component_name]["validations"]["namespace"]
-        print(f"Waiting for {component_name} pods to be ready ...")
-        retcode = kubectl_wait_pods(common_label, namespace, identifier)
-        assert retcode == 0
-    print(f"All {component_name} pods are running!")
+        
+        if "validations" in path_dic[component_name]:
+            validate_component_installation(path_dic, component_name)
+        print(f"All {component_name} pods are running!")
+   
+@retry (stop_max_attempt_number=3)
+def validate_component_installation(path_dic,component_name):
+    identifier = path_dic[component_name]["validations"]["identifier"]
+    common_label = path_dic[component_name]["validations"]["common_label"]
+    namespace = path_dic[component_name]["validations"]["namespace"]
+    print(f"Waiting for {component_name} pods to be ready ...")
+    retcode = kubectl_wait_pods(common_label, namespace, identifier)
+    assert retcode == 0
```

Comment:
what is the retry interval? any reason not to use `wait_for`?

Reply:
this is for resolve the pod delay issue @surajkota run into on Friday

Reply:
ok

Reply:
@jsitu777, @rrrkharse is right, we should retry with a delay and try for max of 3 times before failing

ok to change this later

Reply:
actually, its really simple with the libary
```
@retry(stop_max_attempt_number=3, wait_fixed=3000)
```

lets add this